packages/haul-core/src/webpack/plugins/LooseModePlugin.ts

```diff
@@ -0,0 +1,76 @@
+import webpack from 'webpack';
+import sources from 'webpack-sources';
+import path from 'path';
+import { LooseModeConfig } from '../../types';
+
+/**
+ * Enable JavaScript loose mode, by removing `use strict` directives from the code.
+ * This plugin should only be used for compatibility reasons with Metro, where some libraries
+ * might not work in JavaScript Strict mode.
+ */
+export class LooseModePlugin {
+  shouldUseLoosMode: (filename: string) => boolean;
+
+  /**
+   * Constructs new `LooseModePlugin`.
+   *
+   * @param config - Plugin config value, can be either a:
+   * - `boolean` - enables or disables loose mode for all the modules within a bundle,
+   * - `string[]` - enables loose mode for only modules specified within the array,
+   * - `RegExp[]` - enables loose mode for only modules matching any of the regex within the array,
+   * - `(filename: string) => boolean` - enables loose mode for only modules, for which the function returns `true`.
+   */
+  constructor(config: LooseModeConfig) {
+    if (config === true) {
+      this.shouldUseLoosMode = () => true;
+    } else if (Array.isArray(config)) {
+      this.shouldUseLoosMode = (filename: string) => {
+        return (config as Array<string | RegExp>).some(element => {
+          if (typeof element === 'string') {
+            if (!path.isAbsolute(element)) {
+              throw new Error(
+                `${element} in 'looseMode' property must be an absolute path or regex`
+              );
+            }
+            return element === filename;
+          }
+
+          return element.test(filename);
+        });
+      };
+    } else if (typeof config === 'function') {
+      this.shouldUseLoosMode = config;
+    } else {
+      this.shouldUseLoosMode = () => false;
+    }
+  }
+
+  apply(compiler: webpack.Compiler) {
+    compiler.hooks.make.tap(
+      'LooseModeWebpackPlugin',
+      (compilation: webpack.compilation.Compilation) => {
+        compilation.moduleTemplates.javascript.hooks.render.tap(
+          'LooseModeWebpackPlugin',
```

Comment:
for consistency
```suggestion
      'LooseModePlugin',
      (compilation: webpack.compilation.Compilation) => {
        compilation.moduleTemplates.javascript.hooks.render.tap(
          'LooseModePlugin',
```

Reply:
Or maybe even prefix this with `Haul`? Not sure how webpack resolves naming conflicts, I supposed there may be various LooseModePlugins out there